sphinx_js/renderers.py

```diff
@@ -115,11 +115,12 @@ def _formal_params(self, doclet):
 
         # Harvest params from the @param tag unless they collide with an
         # explicit formal param. Even look at params that are really
-        # documenting subproperties of formal params.
-        params = reduce(
-            lambda l, v: l + [v] if not v in l else l,
-            [param['name'].split('.')[0] for param in doclet.get('params', [])],
-            [])
+        # documenting subproperties of formal params. Also handles params
+        # default values.
+        params = ["%s=%s" % (param, value) if value else param for param, value in reduce(
```

Comment:
Haha, okay, this needs some intermediate naming to make it readable. :-) Maybe break something out as a function, maybe just do part beforehand and stick it in a variable, but it's a lot to keep in your head as one big blob.

For that matter, this could keep us from recalculating `[i[0] for i in params]` every time around the reduce loop.

Reply:
You are right, it was funny to write but it is unreadable :)